src/pages/similarity-indexes.html

```diff
@@ -43,47 +43,47 @@ <h3 class="mb-0"><span class="collapsible-heading" data-toggle="collapse" data-t
+							<span class="text-muted mr-2" ng-show="similarityIndexStatus.FAILED === index.status" gdb-tooltip="{{'index.building.failed.tooltip' | translate}}">{{'failed.status' | translate}}</span>
+							<span class="text-muted mr-2" ng-show="similarityIndexStatus.INTERRUPTED === index.status" gdb-tooltip="{{'index.building.interrupted.tooltip' | translate}}">{{'interrupted.status' | translate}}</span>
+							<span class="text-muted mr-2" ng-show="similarityIndexStatus.OBSOLETE === index.status" gdb-tooltip="{{'index.obsolete.tooltip' | translate}}">{{'obsolete.status' | translate}}</span>
+							<span class="btn btn-link similarity-index-building-loader" ng-show="similarityIndexStatus.BUILDING === index.status || similarityIndexStatus.REBUILDING === index.status || similarityIndexStatus.CREATING === index.status">
           				        <em class="icon-reload loader"></em>
       				        </span>
 
 							<span class="btn btn-link view-sparql-query-btn" ng-click="viewCreateQuery(index)"
-                                  ng-if="index.selectQuery && (index.type == 'text' || index.type == 'predication' || index.type == 'textLiteral')"
+                                  ng-if="index.selectQuery && (similarityIndexType.TEXT === index.type || similarityIndexType.PREDICATION === index.type || similarityIndexType.TEXT_LITERAL === index.type)"
                                   gdb-tooltip="{{'view.sparql.query.title' | translate}}">
 									<em class="icon-sparql"></em>
 							</span>
                             <span class="actions-bar hovered-item">
 								<a class="btn btn-link edit-query-btn"
-                                   ng-if="canEditRepo && index.status != 'BUILDING' && index.status != 'REBUILDING' && index.status != 'CREATING'" href="similarity/index/create?selectQuery={{encodeURIComponent(index.selectQuery)}}&options={{index.options}}&editSearchQuery&name={{index.name}}&infer={{index.infer}}&sameAs={{index.sameAs}}&stopList={{index.stopList}}&type={{index.type}}&analyzer={{index.analyzer}}&searchQuery={{index.searchQuery ? encodeURIComponent(index.searchQuery) : ''}}&analogicalQuery={{index.analogicalQuery ? encodeURIComponent(index.analogicalQuery) : ''}}"
-                                        gdb-tooltip="{{index.type == 'predication' ? 'edit.search.analogical.query.title'
+                                   ng-if="canEditRepo && similarityIndexStatus.BUILDING !== index.status && similarityIndexStatus.REBUILDING !== index.status && similarityIndexStatus.CREATING !== index.status" href="similarity/index/create?selectQuery={{encodeURIComponent(index.selectQuery)}}&options={{index.options}}&editSearchQuery&name={{index.name}}&infer={{index.infer}}&sameAs={{index.sameAs}}&stopList={{index.stopList}}&type={{index.type}}&analyzer={{index.analyzer}}&searchQuery={{index.searchQuery ? encodeURIComponent(index.searchQuery) : ''}}&analogicalQuery={{index.analogicalQuery ? encodeURIComponent(index.analogicalQuery) : ''}}"
```

Comment:
This thing is a mess :(

Reply:
Yes I agree. I will clean it. But if I make everything in one MR it will be big as before.

Reply:
yeah, its OK, I just comment on things that we should consider